airflow/contrib/operators/gcp_translate_speech_operator.py

```diff
@@ -0,0 +1,151 @@
+
+    :param audio: audio data to be recognized. See more:
+        https://googleapis.github.io/google-cloud-python/latest/speech/gapic/v1/types.html#google.cloud.speech_v1.types.RecognitionAudio
+    :type audio: dict or google.cloud.speech_v1.types.RecognitionAudio
+
+    :param config: information to the recognizer that specifies how to process the request. See more:
+        https://googleapis.github.io/google-cloud-python/latest/speech/gapic/v1/types.html#google.cloud.speech_v1.types.RecognitionConfig
+    :type config: dict or google.cloud.speech_v1.types.RecognitionConfig
+
+    :type target_language: str
+    :param target_language: The language to translate results into. This
+      is required by the API and defaults to
+      the target language of the current instance.
+
+    :type format_: str or None
+    :param format_: (Optional) One of ``text`` or ``html``, to specify
+      if the input text is plain text or HTML.
+
+    :type source_language: str or None
+    :param source_language: (Optional) The language of the text to
+      be translated.
+
+    :type model: str or None
+    :param model: (Optional) The model used to translate the text, such
+      as ``'base'`` or ``'nmt'``.
+
+    :param project_id: Optional, Google Cloud Platform Project ID where the Compute
+        Engine Instance exists.  If set to None or missing, the default project_id from the GCP connection is
+        used.
+    :type project_id: str
+
+    :param gcp_conn_id: Optional, The connection ID used to connect to Google Cloud
+        Platform. Defaults to 'google_cloud_default'.
+    :type gcp_conn_id: str
+    """
+
+    # [START translate_speech_template_fields]
+    template_fields = ('target_language', 'format_', 'source_language', 'model', 'project_id', 'gcp_conn_id')
+    # [END translate_speech_template_fields]
+
+    @apply_defaults
+    def __init__(
+        self,
+        audio,
+        config,
+        target_language,
+        format_,
+        source_language,
+        model,
+        project_id=None,
+        gcp_conn_id='google_cloud_default',
+        *args,
+        **kwargs
+    ):
+        super(GcpTranslateSpeechOperator, self).__init__(*args, **kwargs)
+        self.audio = audio
+        self.config = config
+        self.target_language = target_language
+        self.format_ = format_
+        self.source_language = source_language
+        self.model = model
+        self.project_id = project_id
+        self.gcp_conn_id = gcp_conn_id
+
+    def execute(self, context):
+        _speech_to_text_hook = GCPSpeechToTextHook(gcp_conn_id=self.gcp_conn_id)
+        _translate_hook = CloudTranslateHook(gcp_conn_id=self.gcp_conn_id)
+
+        recognize_result = _speech_to_text_hook.recognize_speech(
+            config=self.config, audio=self.audio
+        )
+        recognize_dict = MessageToDict(recognize_result)
+
+        try:
+            transcript = recognize_dict['results'][0]['alternatives'][0]['transcript']
```

Comment:
I think we should explicitly handle case when there are no results - i.e. results is []. In this case we should return "no results" result rather than fail (which will be the current behaviour I think).